hexrd/ui/mask_manager_dialog.py

```diff
@@ -84,8 +84,10 @@ def setup_connections(self):
         self.ui.export_masks.clicked.connect(self.export_visible_masks)
         HexrdConfig().mode_threshold_mask_changed.connect(
             self.update_masks_list)
+        HexrdConfig().detectors_changed.connect(self.clear_masks)
 
     def setup_table(self, status=True):
+        self.ui.masks_table.blockSignals(True)
```

Comment:
I think blocking signals should be done with a context manager. You can use `block_signals` from `hexrd.ui.utils` for this, like so:
```python
with block_signals(self.ui.masks_table):
    ...
```

Although exceptions are undesirable and hopefully not going to occur, they do happen sometimes, and I think it is best to keep hexrdgui in a usable state (if possible) when they do.

Without the context manager, if an exception were raised in the body of this function, the signals from the table would remain blocked as the user tried to continue using the program (since the code to unblock signals would never be reached).

Also, the context manager restores the previously blocked state once it is finished (rather than just setting `blockSignals(False)`), which can be important if a function higher in the call stack already blocked the signals and expects the signals to remain blocked (and, if the signals become unblocked, this can result in a hard-to-track bug). This may not be important in this specific case, but it makes the code easier to maintain in case we do something like that in the future.

Reply:
Thank you, this makes sense. Returning to the previous state is a good point and not one I was thinking of, thanks for explaining everything! Added in.